### PR TITLE
AP_Scripting: change some _INFO msgs to _CRITICAL

### PR DESCRIPTION
**Pull request by magicrub**
Some of the AP_Scripting error handling gcs send_texts as MAV_SEVERITY_INFO deserve to get bumped up to MAV_SEVERITY_CRITICAL

### CODE REVIEW

File: libraries/AP_Scripting/AP_Scripting.cpp [hunk 1]
```diff
@@ -316,7 +316,7 @@ void AP_Scripting::handle_mission_command(const AP_Mission::Mission_Command& cmd
             mission_data = nullptr;
         }
         if (mission_data == nullptr) {
-            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Scripting: %s", "unable to receive mission command");
+            GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "Scripting: %s", "unable to receive mission command");
```
Comment (peterbarker) [inline libraries/AP_Scripting/AP_Scripting.cpp:319]: This one might not deserve this.  Not in immediate chance of crashing here...

Comment (tridge) [reply libraries/AP_Scripting/AP_Scripting.cpp:319]: use ERROR instead

**Review by tridge — CHANGES_REQUESTED** [review]
add SCR_NUM_EXPECTED for the number of expected scripts, and pre-arm if you don't load at least that many would cover this better

**Comment by magicrub** [comment]
Adding a new Param here sounds like feature creep

File: libraries/AP_Scripting/AP_Scripting.cpp (hunk 1 above)
Comment (magicrub) [reply libraries/AP_Scripting/AP_Scripting.cpp]: changed to _ERROR

**Comment by magicrub** [comment]
@tridge I don't like the idea of  SCR_NUM_EXPECTED because I have some scripts that totally exit depending on certain criteria. For example: is_VTOL, is_SITL, home/mission is in certain locations, hardware config, attached sensors detected/config.

**Comment by tridge** [comment]
> I don't like the idea of SCR_NUM_EXPECTED because I have some scripts that totally exit depending on certain criteria. For example: is_VTOL, is_SITL, home/mission is in certain locations, hardware config, attached sensors detected/config.

I believe @IamPete1 is adding 2 params, number of scripts found, and number running

**Comment by magicrub** [comment]
All of which are nice-ish ideas but out of scope for this PR.

File: libraries/AP_Scripting/lua_scripts.cpp [hunk 2]
```diff
@@ -240,7 +240,7 @@ void lua_scripts::load_all_scripts_in_dir(lua_State *L, const char *dirname) {
 
     auto *d = AP::FS().opendir(dirname);
     if (d == nullptr) {
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Lua: open directory (%s) failed", dirname);
+        GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "Lua: open directory (%s) failed", dirname);
```
Comment (rmackay9) [inline libraries/AP_Scripting/lua_scripts.cpp:243]: As discussed, let's make this one an ERROR

File: libraries/AP_Scripting/lua_scripts.cpp [hunk 3]
```diff
@@ -240,7 +240,7 @@ void lua_scripts::load_all_scripts_in_dir(lua_State *L, const char *dirname) {
 
     auto *d = AP::FS().opendir(dirname);
     if (d == nullptr) {
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Lua: open directory (%s) failed", dirname);
```
Comment (peterbarker) [inline libraries/AP_Scripting/lua_scripts.cpp:243]: Let's make this WARNING

**Review by peterbarker — CHANGES_REQUESTED** [review]
Just one change

File: libraries/AP_Scripting/lua_scripts.cpp (hunk 3 above)
Comment (magicrub) [reply libraries/AP_Scripting/lua_scripts.cpp:243]: fixed

**Review by peterbarker — APPROVED** [review]

**Pull request merged**